shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/SingleTableRule.java

```diff
@@ -56,11 +57,18 @@
     
     private final Map<String, SingleTableDataNode> singleTableDataNodes;
     
+    private final Map<String, Collection<DataNode>> dataNodeMap;
+    
+    private final Map<String, String> tableNameMap;
```

Comment:
@tuichenchuxin Maybe it is better to define it as a collection.

Reply:
I need to use concurrent map, And because when we drop table,may be the name is not the same case as the actual table, So I use the low case as the key.